mllib/src/test/scala/org/apache/spark/ml/PredictorSuite.scala

```diff
@@ -55,14 +55,50 @@ class PredictorSuite extends SparkFunSuite with MLlibTestSparkContext {
       predictor.fit(df.select(col("label"), col("weight").cast(StringType), col("features")))
     }
   }
+
+  test("multiple columns for features should work well without side effect") {
+    // Should fail due to not supporting multiple columns
+    intercept[IllegalArgumentException] {
+      new MockPredictor(false).setFeaturesCol(Array("feature1", "feature2", "feature3"))
+    }
+
+    // Only use multiple columns for features
+    val df = spark.createDataFrame(Seq(
+      (0, 1, 0, 2, 3),
+      (1, 2, 0, 3, 9),
+      (0, 3, 0, 2, 6)
+    )).toDF("label", "weight", "feature1", "feature2", "feature3")
+
+    val predictor = new MockPredictor().setWeightCol("weight")
+      .setFeaturesCol(Array("feature1", "feature2", "feature3"))
+    predictor.fit(df)
+
+    // Should fail due to wrong type for column "feature1" in schema
+    intercept[IllegalArgumentException] {
+      predictor.fit(df.select(col("label"), col("weight"),
+        col("feature1").cast(StringType), col("feature2"), col("feature3")))
+    }
+
+    val df2 = df.toDF("label", "weight", "features", "feature2", "feature3")
+    // Should fail due to missing "feature1" in schema
+    intercept[IllegalArgumentException] {
+      predictor.setFeaturesCol(Array("feature1", "feature2", "feature3")).fit(df2)
+    }
+
+    // Should fail due to wrong type in schema for single column of features
```

Comment:
I'm don't understand what this one is testing.  what type is wrong here?  feature2 and feature3 look like ints

Reply:
This is for the case users are specifying both single column and multiple columns for features. `features` (The default value) now is used as the single column name so its type should be Vector, but actually is Int .

Reply:
oh, I see, so can you set the original setFeatureCol(String) to "" if you wanted to use "features" in the setFeaturesCol(Array)?  It looks like it but wanted to double check.

Can you update the comment to say something like your explanation.

Reply:
Thanks for review.  Updated the comments
Actually that's expected. I mean only the names ("feature2", "feature3") passed into `setFeaturesCol(Array)` are wanted to use as multiple columns. But "features" is provided in "df2" schema, equal to the default value of the single column name (just like calling `setFeaturesCol("features")`). Then my current design supposes users are trying to use both single column and multiple columns, and does type check for both of them. As said above ,"features" now is used as single column, and should be "Vector" but actually "Int", so the test fails.